woocommerce/woocommerce-gateway-stripe
# Fix double order note issue

Fixes #2463 #2331

## Changes proposed in this Pull Request:
- Lock the order before creating and confirming the intent. Details [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3331#discussion_r1700615706). 

## Testing instructions
- Checkout to `develop` branch.
- Ensure that webhooks are configured properly. 
- Use jurassic tube or ngrok for making your site publicly accessible (do not use `localhost`, I can not reproduce the issue in localhost)
- As a shopper, purchase a product with a card.
- Now check the order in your admin dashboard. You will find 2 order notes about the charge and order status transition.

<img width="411" alt="Screenshot 2024-08-02 at 12 22 14 AM" src="https://github.com/user-attachments/assets/85290287-39d1-474a-bf0e-fa751e2d77e3">

- Now checkout to this branch `fix/double-processing`. 
- As a shopper, purchase a product with a card again.
- Now check the order in your admin dashboard. This time the order should not have any duplicate notes.
